numpy/random/mtrand/mtrand.pyx

```
@@ -4653,6 +4673,191 @@ cdef class RandomState:
+
+        # The C code that implements the 'count' method will malloc an
+        # array of size total*sizeof(npy_intp). Here we ensure that that
+        # product does not overflow.
+        max_long_index = NPY_MAX_LONG / sizeof(npy_intp)
+        if method == 'count' and total > max_long_index:
+            raise ValueError("When method is 'count', sum(colors) must not "
+                             "exceed %d" % max_long_index)
+        if nsample > total:
+            raise ValueError("nsample > sum(colors)")
+
+        if num_colors == 0:
+            if size is None:
+                return np.array([], dtype=int)
+            else:
+                return np.empty(size + (0,), dtype=int)
+
+        shape = _shape_from_size(size, num_colors)
+        sample = np.zeros(shape, dtype=np.int_)
+        sample_ptr = <long*> PyArray_DATA(sample)
+        sample_size = sample.size
+
+        if method == "count":
+            with self.lock, nogil:
+                result = multivar_hypergeom_count(self.internal_state,
+                                                  total,
+                                                  num_colors, colors_ptr,
+                                                  nsample,
+                                                  sample_size, sample_ptr)
+            if result == -1:
+                raise MemoryError("Insufficent memory for multivariate_"
+                                  "hypergeometric with method='count' and "
+                                  "sum(colors)=%d" % total)
+        else:
+            with self.lock, nogil:
+                result = multivar_hypergeom_iterative(self.internal_state,
+                                                      total,
+                                                      num_colors, colors_ptr,
+                                                      nsample,
+                                                      sample_size, sample_ptr)
+
```

Might want to raise an error if method is not in `['count', 'iterative']`.

---

NVM.